src/language-yaml/utils.js

```diff
@@ -282,7 +282,7 @@ function getBlockValueLineContents(
 
     let trailingNewlineCount = 0;
     for (let i = lineContents.length - 1; i >= 0; i--) {
-      if (lineContents[i].length === 0) {
+      if (lineContents[i].every((line) => line.trim() === "")) {
```

Comment:
I'm not sure this is the correct fix.

But we need test all these case, and more

```
runFormatTest(
  {
    importMeta: import.meta,
    snippets: [
		'foo: |',
		'foo: |\n',
		'foo: |\n\n',
		'foo: |\n\n\n',
		'foo: |  ',
		'foo: |  \n',
		'foo: |  \n\n',
		'foo: |  \n\n\n',
		'foo: >',
		'foo: >\n',
		'foo: >\n\n',
		'foo: >\n\n\n',
		'foo: >  ',
		'foo: >  \n',
		'foo: >  \n\n',
		'foo: >  \n\n\n',
	],
  },
  ["yaml"],
);
```